<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>COMDeviceBase</class>
 <widget class="QWidget" name="COMDeviceBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>635</width>
    <height>366</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout" stretch="50,50">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>receive</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="1" column="0" colspan="2">
       <widget class="QTextBrowser" name="textBrowser_device_rec"/>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>format</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="comboBox_device_rec_format"/>
      </item>
      <item row="2" column="1">
       <widget class="QPushButton" name="pushButton_device_rec_clear">
        <property name="text">
         <string>clear</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>send</string>
     </property>
     <layout class="QGridLayout" name="gridLayout" columnstretch="10,0,0">
      <item row="0" column="0" rowspan="5" colspan="2">
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>current command</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="5" column="1">
          <widget class="QCheckBox" name="checkBox_device_send_disabled">
           <property name="text">
            <string>Disabled</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>count</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLineEdit" name="lineEdit_device_send_position">
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="7" column="1" colspan="2">
          <widget class="QPushButton" name="pushButton_device_send_open_robot_list">
           <property name="text">
            <string>choose from robot list</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QLineEdit" name="lineEdit_device_send_period"/>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>period (ms)</string>
           </property>
          </widget>
         </item>
         <item row="8" column="1" colspan="2">
          <widget class="QPushButton" name="pushButton_device_send_update">
           <property name="text">
            <string>update</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>position</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1" colspan="2">
          <widget class="QLineEdit" name="lineEdit_device_send"/>
         </item>
         <item row="6" column="1">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="3" column="2">
          <widget class="QComboBox" name="comboBox_device_send_format"/>
         </item>
         <item row="9" column="1" colspan="2">
          <widget class="QPushButton" name="pushButton_device_send_add2list">
           <property name="text">
            <string>add to list</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLineEdit" name="lineEdit_device_send_count"/>
         </item>
         <item row="3" column="1">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>format</string>
           </property>
          </widget>
         </item>
         <item row="10" column="1" colspan="2">
          <widget class="QPushButton" name="pushButton_device_send">
           <property name="text">
            <string>Send current</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="2" rowspan="5">
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>command list</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="0" column="0">
          <widget class="QTableWidget" name="tableWidget_device_send_list"/>
         </item>
         <item row="1" column="0">
          <widget class="QPushButton" name="pushButton_device_send_all">
           <property name="text">
            <string>Send all list</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
